File: dsa/lovebabar/neg.py
"""
Move all negative numbers to beginning and positive to end with constant extra space

    Difficulty Level : Easy
    Last Updated : 09 Jun, 2021

An array contains both positive and negative numbers in random order. Rearrange the array elements so that all negative numbers appear before all positive numbers.

Examples : 

Input: -12, 11, -13, -5, 6, -7, 5, -3, -6
Output: -12 -13 -5 -7 -3 -6 11 6 5

Note: Order of elements is not important here.


"""

inp = [-1, 0, 5, 7, 3, 45, 5, 56, 7, 6, -2, -4, 8, -3]

for i in range(len(inp)):
    if(inp[i] < 0):
        x = inp.pop(i)
        inp.insert(0, x)
print(inp)

File: dsa/lovebabar/neg.cpp
//Move all the negative elements to one side of the array

#include <bits/stdc++.h>

using namespace std;

vector<int> shuffler(vector<int> a, int l, int h)
{
    int temp;
    while (l <= h)
    {
        if (a[l] > 0 && a[h] < 0)
        {
            temp = a[h];
            a[h] = a[l];
            a[l] = a[h];
            l++;
            h--;
        }
        else if (a[l] < 0 || (a[h] < 0))
            l++;
        else if (a[h] > 0 || a[l] > 0)
            h--;
    }
    return a;
}

int main()
{

    int n, j;
    cin >> n;
    vector<int> a;
    for (j = 0; j < n; j++)
        cin >> a[j];
    int l = 0, h = n - 1;
    a = shuffler(a, l, h);
    for (j = 0; j < n; j++)
        cout << a[j] << "\t";
    cout << endl;
}